src/auth/featureFlagStorage.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2024 PixieBrix, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { getApiClient } from "@/data/service/apiClient";
+import { type components } from "@/types/swagger";
+import { once } from "lodash";
+import { addListener } from "@/auth/authStorage";
+import { CachedFunction } from "webext-storage-cache";
+
+const featureFlags = new CachedFunction("getFeatureFlags", {
+  async updater(): Promise<string[]> {
+    const client = await getApiClient();
+    const { data, status, statusText } =
+      await client.get<components["schemas"]["Me"]>("/api/me/");
+
+    if (status >= 400) {
+      console.debug(`Failed to fetch feature flags: ${status} ${statusText}`);
+      return [];
+    }
+
+    return [...(data.flags ?? [])];
```

Comment:
This it's an API response, we shouldn't need to clone the array 

```suggestion
    return data.flags ?? [];
```

Reply:
It's `readonly` in the swagger type, so was doing this to convert to a normal array 🤷‍♂️

Reply:
I see, I think you can change the return type to `readonly` too, in case you want to remove this